clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -512,12 +495,10 @@ private void completeBatch(ProducerBatch batch, ProduceResponse.PartitionRespons
                 final RuntimeException exception;
                 if (error == Errors.TOPIC_AUTHORIZATION_FAILED)
                     exception = new TopicAuthorizationException(batch.topicPartition.topic());
+                else if (error == Errors.CLUSTER_AUTHORIZATION_FAILED)
+                    exception = new ClusterAuthorizationException("The producer is not authorized to do idempotent sends");
```

Comment:
It would be better to use the `Errors.CLUSTER_AUTHORIZATION_FAILED.execption()` here and elsewhere. THis way the error messages can be updated consistently everywhere for the same error type.

I have already started doing this for the `ProducerFencedException` in an upcoming patch.